yarpcerrors/codes.go

```diff
@@ -67,10 +73,12 @@ const (
 	// instead for those errors). `PermissionDenied` must not be
 	// used if the caller can not be identified (use `Unauthenticated`
 	// instead for those errors).
+	// This is considered as a client error.
 	CodePermissionDenied Code = 7
 
 	// CodeResourceExhausted means some resource has been exhausted, perhaps a per-user
 	// quota, or perhaps the entire file system is out of space.
+	// This is considered as a client error.
```

Comment:
Shouldn't resource exhausted be a server error? This could be server's fault running at incorrect capacity?

Reply:
It is the kind of error which can be the fault of both server and client.

See https://grpc.github.io/grpc/core/md_doc_statuscodes.html